src/common/misc.c

```diff
@@ -662,8 +663,11 @@ daos_prop_free(daos_prop_t *prop)
 		switch (entry->dpe_type) {
 		case DAOS_PROP_PO_LABEL:
 		case DAOS_PROP_CO_LABEL:
-			if (entry->dpe_str)
-				D_FREE(entry->dpe_str);
+			D_FREE(entry->dpe_str);
```

Comment:
In the C standard free(NULL) is a no-op. I checked the definition of D_FREE to see if it adds any logic that requires the param to be non-null; it doesn't. So the null check should not be necessary. I added a similar case below, so I updated this one to be consistent.

Reply:
I know free(NULL) works. :) A couple of years ago, for debugging purposes, I had to convert all cases in rdb that might free NULLs into "if not NULL" checks. Hence, it seems we should follow Xuezhao's check here. But anyway, no need to update the patch for this.

Reply:
I'll re-add the checks since I am also adding Xuezhao's suggestion below. Wondering if it might be useful to move the check into the D_FREE macro instead, if it is needed everywhere we use D_FREE.